### DISCUSSION TITLE
[Bug or FR?] Renovate doesn't update indirect dependencies in peer go modules

### DISCUSSION THREAD

**Discussion by eriksw** [renovatebot/renovate#12998] [Request Help] [answered]
Not sure whether this should be opened as a bug report or a feature request:

It appears that renovate doesn't track/act appropriately on peer go module relationships, specifically, when it updates a _direct_ dependency of one go module, it doesn't update the _indirect_ dependencies of the go modules that depend on the updated module.

Scenario:

    monorepo/go/a:
        has a dependency that renovate is updating (libfoo)
    monorepo/go/b:
        has a dependency on monorepo/go/a
        has a replace rule for monorepo/go/a => ../a
        does not directly import "libfoo"
        has "indirect" go.mod and go.sum entries for libfoo because go adds/requires them now

In this scenario, when updating libfoo in `monorepo/go/a/...` I need renovate to also update `monorepo/go/b/go.*` at the same time, in the same commit.

Because renovate currently does not do this, the PR to update module a fails because we have a test that ensures our module files are always tidy. That fails because when the test runs `go mod tidy` in b, it is stale and updated/rewritten by the `go mod tidy` invocation in b, which updates b's indirect dependencies in go.mod and go.sum.

**Comment by rarkins** [comment, marked as answer]
Please raise a feature request with reproduction repo

**Reply by eriksw** [reply]
Created: https://github.com/renovatebot/renovate/issues/12999

Aside: The "Create issue from discussion" lost the relationship to the discussion when I chose the Feature Request option.
